src/cpu/flexcpu/simple_dataflow_thread.cc

```diff
@@ -447,7 +451,23 @@ SDCPUThread::issueInstruction(weak_ptr<InflightInst> inst)
         // If we are still running and know where to fetch, we should
         // immediately send the necessary requests to the CPU to fetch the next
         // instruction
-        advanceInst();
+        advanceInst(getNextPC());
+    } else if (_cpuPtr->hasBranchPredictor()) {
```

Comment:
Should this be part of the `getNextPC()` logic instead?

Reply:
Well it can't be if we want to be able to have access time to the predictor, since it would be a in a different call stack. I think if you didn't want it there, it'd have to be in some variant of advanceinst, which would have to schedule a call in a different call stack in the branch prediction case.